check-dependencies.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+set -euo pipefail
+
+# Testing with this to simulate different installed apps:
+# docker run -it -v ~/Source/aws-cdk:/var/cdk ubuntu bash
+
```

Comment:
BTW, `docker` is also a prerequisite. It's needed for @aws-cdk/aws-s3-assets tests

Reply:
Added a check for docker >= 19.03